File: examples/rails/app/models/post.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: posts
#
#  id         :uuid             not null, primary key
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  text       :string
#  parent_id  :uuid
#  author_id  :uuid
#
class Post < ApplicationRecord
  belongs_to :author, class_name: "User"
  belongs_to :parent, optional: true, class_name: "Post", touch: true

  has_many :comments, dependent: :destroy, foreign_key: :parent_id, class_name: "Post", inverse_of: :parent
  has_many :reactions, dependent: :destroy

  validates :text, presence: true

  scope :root, -> { where(parent: nil) }
end

File: examples/go/graph/model/post.go
package model

import "database/sql"

type Post struct {
	ID        string          `json:"id" gorm:"type:uuid;default:gen_random_uuid()"`
	CreatedAt ISO8601DateTime `json:"createdAt" gorm:"not null"`
	UpdatedAt ISO8601DateTime `json:"updatedAt" gorm:"not null"`
	AuthorID  string          `json:"authorId" gorm:"not null"`
	Author    *User           `json:"author"`
	ParentID  sql.NullString  `json:"parentId"`
	Parent    *Post           `json:"parent,omitempty"`
	Text      string          `json:"text" gorm:"not null"`
	Reactions []*Reaction     `json:"reactions"`
}
